### PR TITLE
payment-detail, coming-soon, initial facebook sign up, review form

### PR DESCRIPTION
**Pull request by kru**
screen shot 2018-02-28 at 23 31 57

screen shot 2018-02-28 at 23 31 30

screen shot 2018-03-01 at 00 05 52

### CODE REVIEW

File: mobile/src/views/login-signup/signup/index.js [hunk 1]
```diff
@@ -13,11 +14,30 @@ const styles = StyleSheet.create({
   },
 })
 
+async function fbSignUp() {
+  const { type, token } = await Expo.Facebook.logInWithReadPermissionsAsync(
+    '337883893384348',
```
Comment (EdmundMai) [inline mobile/src/views/login-signup/signup/index.js]: put this in a config file

File: mobile/src/views/login-signup/index.js [hunk 2]
```diff
@@ -155,7 +155,10 @@ class Wrapper extends React.Component {
             justifyContent: 'space-around',
           }}
         >
-          <TouchableOpacity style={styles.signUpFacebook}>
+          <TouchableOpacity
+            style={styles.signUpFacebook}
+            onPress={() => this.props.onPress()}
```
Comment (EdmundMai) [inline mobile/src/views/login-signup/index.js]: `onPress={this.props.onPress}`

File: mobile/src/views/coming-soon/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+const ComingSoon = () => (
+  <View style={styles.container}>
+    <View
+      style={{
+        flex: 0.2,
+        width: '80%',
+        alignItems: 'center',
+        justifyContent: 'center',
+      }}
```
Comment (EdmundMai) [inline mobile/src/views/coming-soon/index.js]: put the styles in a style object

File: mobile/src/views/login-signup/signup/index.js [hunk 4]
```diff
@@ -13,11 +14,30 @@ const styles = StyleSheet.create({
   },
 })
 
+async function fbSignUp() {
```
Comment (EdmundMai) [inline mobile/src/views/login-signup/signup/index.js]: this should be in an epic

File: mobile/src/redux/epics/signup-epic.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import { Observable } from 'rxjs'
+
+import { SIGNUP_ACTIONS, signUpActions } from '../modules/signup'
+import { signUp } from '../effects/facebook'
+
+export const signUpRequest = action$ =>
+  action$.ofType(SIGNUP_ACTIONS.REQUEST).concatMap(() =>
+    Observable.fromPromise(signUp())
+      .flatMap(payload =>
+        ajax
```
Comment (AaronVasquez) [inline mobile/src/redux/epics/signup-epic.js]: Let's use https://github.com/axios/axios for http.

Include this portion in the effects file too.

So you're effects code would look something like:
```js
export const signUp = () =>
    Expo.Facebook.logInWithReadPermissionsAsync(CONFIG.FACEBOOK_APP_ID, {
        permissions: ['public_profile', 'email'],
     })
    .then(payload => axios.get(`https://graph.facebook.com/me?access_token=${payload.token}`))
```

epic code:
```js
Observable.fromPromise(signUp())
    .switchMap(signUpActions.fulfilled)
    .catch(error => Observable.of(signUpActions.rejected(error)))
```

File: mobile/src/redux/effects/review.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import Axios from 'axios'
+
+export const fetchReview = () =>
```
Comment (AaronVasquez) [inline mobile/src/redux/effects/review.js]: This will probably be `/effects/api.js`

File: mobile/src/redux/effects/facebook.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import Expo from 'expo'
```
Comment (AaronVasquez) [inline mobile/src/redux/effects/facebook.js]: Nit: Makes `Expo` and `Axios` lowercase (Same everywhere else). Uppercase is only for comopnents and classes.

File: mobile/src/redux/epics/review-epic.js [hunk 8]
```diff
@@ -1,24 +1,18 @@
...
-      ajax
-        .getJSON('https://jsonplaceholder.typicode.com/posts/1')
-        .flatMap(() =>
-          Observable.of(writeReviewActions.fetchReviewFulfilled(data)),
-        )
-        .catch(({ xhr }) =>
-          Observable.of(writeReviewActions.fetchReviewRejected(xhr.response)),
-        ),
-    ),
+  action$.ofType(WRITE_REVIEW_ACTIONS.FETCH_REVIEW).switchMap(() =>
```
Comment (AaronVasquez) [inline mobile/src/redux/epics/review-epic.js]:
```js
export const fetchReviewEpic = action$ =>
  action$.ofType(WRITE_REVIEW_ACTIONS.FETCH_REVIEW).switchMap(action =>
    Observable.fromPromise(fetchReview())
      .map(writeReviewActions.fetchReviewFulfilled)
      .catch(err => Observable.of(writeReviewActions.fetchReviewRejected(error.message)))
  )
```

File: mobile/src/redux/epics/signup-epic.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { Observable } from 'rxjs'
+
+import { SIGNUP_ACTIONS, signUpActions } from '../modules/signup'
+import { signUp } from '../effects/facebook'
+
+export const signUpRequest = action$ =>
```
Comment (AaronVasquez) [inline mobile/src/redux/epics/signup-epic.js]:
```js
export const signUpRequest = action$ =>
  action$.ofType(SIGNUP_ACTIONS.REQUEST).switchMap(action =>
    Observable.fromPromise(signUp())
      .map(signUpActions.fulfilled)
      .catch(err => Observable.of(signUpActions.rejected(err))),
  )
```

**Comment by AaronVasquez** [comment]
@EdmundMai You have a blocking review. Can you please review when you have time?

**Pull request merged**